ext/reflection/php_reflection.c

```diff
@@ -7207,6 +7208,70 @@ static zval *_reflection_write_property(zend_object *object, zend_string *name,
 }
 /* }}} */
 
+ZEND_METHOD(ReflectionConstant, __construct)
+{
+	zend_string *const_name;
+
+	zval *object = ZEND_THIS;
+	reflection_object *intern = Z_REFLECTION_P(object);
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_STR(const_name)
+	ZEND_PARSE_PARAMETERS_END();
+
+	zval *const_zv = zend_get_constant(const_name);
+	if (!const_zv) {
+		zend_throw_exception_ex(reflection_exception_ptr, 0, "Constant \"%s\" does not exist", ZSTR_VAL(const_name));
+		RETURN_THROWS();
+	}
+
+	zend_constant *const_ = (zend_constant*)((uintptr_t)const_zv - XtOffsetOf(zend_constant, value));
+
+	intern->ptr = const_;
+	intern->ref_type = REF_TYPE_OTHER;
+
+	ZVAL_STR_COPY(reflection_prop_name(object), const_name);
```

Comment:
Calling the constructor twice will leak the const name:

```
<?php
$name = bin2hex(random_bytes(4)); // name is not interned
define($name, 'test');
$r = new ReflectionConstant($name);
$r->__construct($name);
```

Reply:
Throwing would be fine with me, but maybe not on existing classes as some code may rely on this.

Some classes avoid the leak, but not all.